larpandoracontent/LArHelpers/LArGeometryHelper.cc

```diff
@@ -331,31 +331,64 @@ CartesianVector LArGeometryHelper::ProjectDirection(const Pandora &pandora, cons
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
-float LArGeometryHelper::GetWireZPitch(const Pandora &pandora)
+float LArGeometryHelper::GetWireZPitch(const Pandora &pandora, const float maxWirePitchWDiscrepancy)
 {
-    return (pandora.GetGeometry()->GetLArTPC().GetWirePitchW());
+    const LArTPCMap &larTPCMap(pandora.GetGeometry()->GetLArTPCMap());
+
+    if (larTPCMap.empty())
+    {
+        std::cout << "LArGeometryHelper::GetWireZPitch - LArTPC description not registered with Pandora as required " << std::endl;
+        throw StatusCodeException(STATUS_CODE_NOT_INITIALIZED);
+    }
+
+    const LArTPC *const pFirstLArTPC(larTPCMap.begin()->second);
+    const float wirePitchW(pFirstLArTPC->GetWirePitchW());
+
+    for (const LArTPCMap::value_type &mapEntry : larTPCMap)
+    {
+        const LArTPC *const pLArTPC(mapEntry.second);
+
+        if (std::fabs(wirePitchW - pLArTPC->GetWirePitchW()) > maxWirePitchWDiscrepancy)
+        {
+            std::cout << "LArGeometryHelper::GetWireZPitch - Plugin does not support provided LArTPC configurations " << std::endl;
```

Comment:
Pedantic. Does it make sense for this message to use the term Plugin? Maybe. Probably not quite.